stacks-signer/src/v0/signer.rs

```diff
@@ -233,23 +244,105 @@ impl Signer {
+        let mut block_info = BlockInfo::from(block_proposal.clone());
+
+        // Get sortition view if we don't have it
+        if sortition_state.is_none() {
+            *sortition_state = SortitionsView::fetch_view(stacks_client)
+                .inspect_err(|e| {
+                    warn!(
+                        "{self}: Failed to update sortition view: {e:?}";
+                        "signer_sighash" => %signer_signature_hash,
+                        "block_id" => %block_proposal.block.block_id(),
+                    )
+                })
+                .ok();
+        }
+
+        // Check if proposal can be rejected now if not valid agains sortition view
+        let block_response = if let Some(sortition_state) = sortition_state {
+            match sortition_state.check_proposal(
+                stacks_client,
+                &self.signer_db,
+                &block_proposal.block,
+                miner_pubkey,
+            ) {
+                // Error validating block
+                Err(e) => {
+                    warn!(
+                        "{self}: Error checking block proposal: {e:?}";
+                        "signer_sighash" => %signer_signature_hash,
+                        "block_id" => %block_proposal.block.block_id(),
+                    );
+                    Some(BlockResponse::rejected(
+                        block_proposal.block.header.signer_signature_hash(),
+                        RejectCode::ConnectivityIssues,
+                    ))
+                }
+                // Block proposal is bad
+                Ok(false) => {
+                    warn!(
+                        "{self}: Block proposal invalid";
+                        "signer_sighash" => %signer_signature_hash,
+                        "block_id" => %block_proposal.block.block_id(),
+                    );
+                    Some(BlockResponse::rejected(
+                        block_proposal.block.header.signer_signature_hash(),
+                        RejectCode::ValidationFailed(ValidateRejectCode::InvalidBlock),
+                    ))
+                }
+                // Block proposal passed check, still don't know if valid
+                Ok(true) => None,
+            }
         } else {
-            debug!(
-                "{self}: received a block proposal for a new block. Submit block for validation. ";
+            warn!(
+                "{self}: Cannot validate block, no sortition view";
                 "signer_sighash" => %signer_signature_hash,
                 "block_id" => %block_proposal.block.block_id(),
             );
-            let block_info = BlockInfo::from(block_proposal.clone());
-            crate::monitoring::increment_block_proposals_received();
+            Some(BlockResponse::rejected(
+                block_proposal.block.header.signer_signature_hash(),
+                RejectCode::ValidationFailed(ValidateRejectCode::InvalidBlock),
```

Comment:
It might be worthwhile having a specific reject code for this but this is non-blocking. If you feel its worthwhile too, just make a ticket. Otherwise ignore me.

Reply:
Agree, I'll make a ticket

Reply:
#4941